internal/istio/httproute.go

```diff
@@ -0,0 +1,65 @@
+package istio
+
+import (
+	"fmt"
+
+	istioapiv1beta1 "istio.io/api/networking/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+)
+
+func NewHTTPRoute(watcher *v1beta2.Watcher) (*istioapiv1beta1.HTTPRoute, error) {
+	if watcher == nil {
+		return nil, fmt.Errorf("watcher must not be nil: %w", ErrInvalidArgument)
```

Comment:
I suggest moving these checks to another helper function that would only validate the watcher then call it here